src/main/java/org/example/Car.java

```diff
@@ -0,0 +1,69 @@
+package org.example;
+
+public abstract class Car implements Vehicle {
+    protected final double fuelTankSize;
+    protected final String fuelType;
+    protected final int maxGears;
+    protected final double consumptionPer100Km;
+    protected double availableFuel;
+    protected int tireSize;
+    protected String chassisNumber;
+    protected double avearageFuelConsumptionPerTrip;
+    protected int currentGear = 0;
+    protected double totalKm;
+    protected double totalFuelConsumption;
+
+    protected Car(double fuelTankSize, String fuelType, int maxGears, double consumptionPer100Km){
+        this.fuelTankSize = fuelTankSize;
+        this.fuelType = fuelType;
+        this.maxGears = maxGears;
+        this.consumptionPer100Km = consumptionPer100Km;
+        this.availableFuel = fuelTankSize;
+    }
+    @Override
+    public void start(){
+        System.out.println("Starting car");
+        avearageFuelConsumptionPerTrip = consumptionPer100Km;
+    }
+    @Override
+    public void stop() {
+        this.currentGear = 0;
+        System.out.println("Setting current gear to 0");
+        System.out.println("Stopping car");
+        System.out.println("Available fuel " +String.format("%,.2f", availableFuel));
+        System.out.println("Total fuel consumption " + String.format("%,.2f",totalFuelConsumption));
+    }
+    protected void shiftGear(int gear) {
+        if ((gear > maxGears) || (gear < 0)) {
+            System.out.println("Invalid gear");}
+        else {
```

Comment:
you can avoid this else,  (and nesting of another if/else) by turning the if above into a guard condition (add return keyword after "invalid gear" message.

Reply:
nesting also can be avoided by wrapping all the consumption stuff in a method